core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -303,7 +287,294 @@ public static Properties filterAndRemovePrefix(Map<String, String> properties, S
+            // UPDATE
+            sql.setString(1, newMetadataLocation);
+            sql.setString(2, oldMetadataLocation);
+            // WHERE
+            sql.setString(3, catalogName);
+            sql.setString(4, namespaceToString(identifier.namespace()));
+            sql.setString(5, identifier.name());
+            sql.setString(6, oldMetadataLocation);
+            return sql.executeUpdate();
+          }
+        });
+  }
+
+  static int updateTable(
+      JdbcClientPool connections,
+      String catalogName,
+      TableIdentifier tableIdentifier,
+      String newMetadataLocation,
+      String oldMetadataLocation)
+      throws SQLException, InterruptedException {
+    return update(
+        true, connections, catalogName, tableIdentifier, newMetadataLocation, oldMetadataLocation);
+  }
+
+  static int updateView(
+      JdbcClientPool connections,
+      String catalogName,
+      TableIdentifier viewIdentifier,
+      String newMetadataLocation,
+      String oldMetadataLocation)
+      throws SQLException, InterruptedException {
+    return update(
+        false, connections, catalogName, viewIdentifier, newMetadataLocation, oldMetadataLocation);
+  }
+
+  private static Map<String, String> get(
+      boolean isTable, JdbcClientPool connections, String catalogName, TableIdentifier identifier)
+      throws SQLException, InterruptedException {
+    return connections.run(
+        conn -> {
+          Map<String, String> tableOrView = Maps.newHashMap();
+
+          String getTableOrViewSqlStatement = JdbcUtil.getSql(isTable);
+          try (PreparedStatement sql = conn.prepareStatement(getTableOrViewSqlStatement)) {
+            sql.setString(1, catalogName);
+            sql.setString(2, namespaceToString(identifier.namespace()));
+            sql.setString(3, identifier.name());
+            ResultSet rs = sql.executeQuery();
+
+            if (rs.next()) {
+              tableOrView.put(JdbcUtil.CATALOG_NAME, rs.getString(JdbcUtil.CATALOG_NAME));
+              tableOrView.put(
+                  (isTable ? JdbcUtil.TABLE_NAMESPACE : JdbcUtil.VIEW_NAMESPACE),
+                  rs.getString((isTable ? JdbcUtil.TABLE_NAMESPACE : JdbcUtil.VIEW_NAMESPACE)));
+              tableOrView.put(
+                  (isTable ? JdbcUtil.TABLE_NAME : JdbcUtil.VIEW_NAME),
+                  rs.getString((isTable ? JdbcUtil.TABLE_NAME : JdbcUtil.VIEW_NAME)));
+              tableOrView.put(
+                  BaseMetastoreTableOperations.METADATA_LOCATION_PROP,
+                  rs.getString(BaseMetastoreTableOperations.METADATA_LOCATION_PROP));
+              tableOrView.put(
+                  BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP,
+                  rs.getString(BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP));
+            }
+
+            rs.close();
+          }
+
+          return tableOrView;
+        });
+  }
+
+  static Map<String, String> getTable(
```

Comment:
nit: we can remove the prefix get (to be uniform with iceberg coding style), can use fetch if really required.

for getTable, getView, get(), getSql()

Reply:
What would you propose ? IMHO `fetch` is not correct for this methods (it selects a table). `get` makes sense to me.
Maybe `selectTable()` instead of `getTable()` ?